assets/src/pages/room/hooks/useMediaDeviceManager.tsx

```diff
@@ -0,0 +1,89 @@
+import { useCallback, useEffect, useState } from "react";
+
+type MediaDeviceManagerConfig = {
+  askOnMount?: boolean;
+};
+
+const showMediaDevicesPrompt = (constraints: MediaStreamConstraints, onSuccess: () => void, onError: () => void) => {
+  navigator.mediaDevices
+    .getUserMedia(constraints)
+    .then((mediaStream) => {
+      onSuccess();
+      mediaStream.getTracks().forEach((track) => {
+        track.stop();
+      });
+    })
+    .catch(() => {
+      onError();
+    });
+};
+
+export const useMediaDeviceManager = ({ askOnMount }: MediaDeviceManagerConfig = {}) => {
+  const [devices, setDevices] = useState<MediaDeviceInfo[]>([]);
+  const [asked, setAsked] = useState(!askOnMount);
+  const [audioPermissionGranted, setAudioPermissionGranted] = useState<boolean | null>(null);
+  const [videoPermissionGranted, setVideoPermissionGranted] = useState<boolean | null>(null);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  useEffect(() => {
+    navigator.mediaDevices.enumerateDevices().then((mediaDevices) => {
+      setDevices(mediaDevices);
+    });
+  }, []);
+
+  useEffect(() => {
+    if (devices.length === 0) return;
+
+    const emptyId = devices.filter((device) => device.kind === "audioinput").find((device) => device.deviceId === "");
+    setAudioPermissionGranted(!emptyId);
+  }, [devices]);
+
+  useEffect(() => {
+    if (devices.length === 0) return;
+
+    const emptyId = devices.filter((device) => device.kind === "videoinput").find((device) => device.deviceId === "");
+    setVideoPermissionGranted(!emptyId);
+  }, [devices]);
```

Comment:
Have you tested that on Firefox?

Reply:
It does not work 😢 Permission manager doesn't work, but users are able to make a call. 

I made Jira for that: https://membraneframework.atlassian.net/browse/MV-229